app/jenkins-for-jira-ui/src/api/getAllJenkinsServers.ts

```diff
@@ -1,10 +1,77 @@
 import { invoke } from '@forge/bridge';
 import { JenkinsServer } from '../../../src/common/types';
+import {
+	AnalyticsEventTypes,
+	AnalyticsScreenEventsEnum,
+	AnalyticsTrackEventsEnum
+} from '../common/analytics/analytics-events';
+import { AnalyticsClient } from '../common/analytics/analytics-client';
+
+const analyticsClient = new AnalyticsClient();
+
+const sendJenkinsServersAnalytics = async (jenkinsServers: JenkinsServer[]): Promise<void> => {
+	await analyticsClient.sendAnalytics(
+		AnalyticsEventTypes.TrackEvent,
+		AnalyticsTrackEventsEnum.GetServerSuccessJenkinsConfigurationName,
+		{
+			source: AnalyticsScreenEventsEnum.ConfigurationConfiguredStateScreenName
+		}
+	);
+
+	const totalNumberOfServers = jenkinsServers.length;
+	const totalNumberOfServersWithPipelines = jenkinsServers.filter((server: JenkinsServer) =>
+		server.pipelines.length === 0).length;
+	const totalNumberOfServersWithoutPipelines = jenkinsServers.filter((server: JenkinsServer) =>
+		server.pipelines.length > 0).length;
+
+	await analyticsClient.sendAnalytics(
+		AnalyticsEventTypes.TrackEvent,
+		AnalyticsTrackEventsEnum.TotalNumberJenkinsServersName,
+		{
+			source: AnalyticsScreenEventsEnum.ConfigurationConfiguredStateScreenName,
+			totalNumberOfServers
+		}
+	);
+
+	await analyticsClient.sendAnalytics(
```

Comment:
I'm wondering if it makes sense to send this (and the following) events when the total number of servers is 0? To me it looks like this case could skew the picture....

Reply:
I don't think it matters too much at this point. We're trying to get a picture of the biggest failure points for customers. I'm only going to add this metrics to the viewed Jenkins config page so 0 servers will be filtered out naturally. Plus I'm planning to let the analytics to soak for a while and anything that doesn't provide value, I'm going to :killitwithfire: